checkov/terraform/checks/graph_checks/oci/AdministratorUserAssociatedWithAPIKey.yaml

```
@@ -0,0 +1,65 @@
+metadata:
+  id: "PLACEHOLDER"
+  name: "Ensure administrator users are associated with API keys"
+  category: "IAM"
+scope:
+  provider: 'oci'
+definition:
+  or:
+    - and:
+      - resource_types:
+          - oci_identity_group
+        connected_resource_types:
+          - oci_identity_user_group_membership
+        operator: exists
+        cond_type: connection
+      - resource_types:
+          - oci_identity_user_group_membership
+        connected_resource_types:
+          - oci_identity_user
+        operator: exists
+        cond_type: connection
+      - cond_type: attribute
+        attribute: name
+        resource_types:
+          - oci_identity_group
+        operator: equals
+        value: "Administrators"
+      - resource_types:
+          - oci_identity_user
+        connected_resource_types:
+          - oci_identity_api_key
+        operator: exists
+        cond_type: connection
+    - and:
+      - resource_types:
+          - oci_identity_group
+        connected_resource_types:
+          - oci_identity_user_group_membership
+        operator: exists
+        cond_type: connection
+      - resource_types:
+          - oci_identity_user_group_membership
+        connected_resource_types:
+              - oci_identity_user
+        operator: exists
+        cond_type: connection
+      - cond_type: attribute
+        attribute: name
+        resource_types:
+          - oci_identity_group
+        operator: not_equals
+        value: "Administrators"
+    - and:
+      - resource_types:
+          - oci_identity_user_group_membership
+        connected_resource_types:
+          - oci_identity_user
+        operator: not_exists
+        cond_type: connection
```

```suggestion
    - resource_types:
        - oci_identity_user_group_membership
      connected_resource_types:
        - oci_identity_user
      operator: not_exists
      cond_type: connection
```
the `and` is not needed here 🙂

---

you could also merge the other two `and` blocks to one and then do some differentiation in a nested `or` block, but I think your approach is easier to understand 👍

---

Deleted the unnecessary `and` block